chain/client/src/stateless_validation/chunk_validator/mod.rs

```diff
@@ -269,6 +269,9 @@ impl Client {
         // wait for validation to finish.
         self.send_state_witness_ack(&witness, &signer);
 
+        #[cfg(feature = "artificial_witness_size")]
```

Comment:
What would happen if a node with this compile feature sent a witness to a node without. Would the node reject the witness entirely, or would it try to operate on the now-present artificially generated entries and somehow fail at a later date (due to e.g. the fact that `added_artificial_witness_size` not a real account.)

From what I can tell, these artificial witnesses are structurally valid, they just use well-known values in certain fields to differentiate them from the regular data.

Reply:
> What would happen if a node with this compile feature sent a witness to a node without? Would the node reject the witness entirely, or would it try to operate on the now-present artificially generated entries and somehow fail at a later date (due to e.g. the fact that added_artificial_witness_size not a real account.)

I believe that witness validation would fail - `apply_chunk` expects that all transactions are valid and will fail if one of them isn't, so witness validation will fail.